src/admin.rs

```diff
@@ -36,6 +37,17 @@ pub async fn list_manifests<R: RangeBounds<u64>>(
     Ok(serde_json::to_string(&manifests)?)
 }
 
+pub async fn list_checkpoints(
+    path: &Path,
+    object_store: Arc<dyn ObjectStore>,
+) -> Result<String, Box<dyn Error>> {
+    let manifest_store = ManifestStore::new(path, object_store);
+    let Some((_, manifest)) = manifest_store.read_latest_manifest().await? else {
+        return Err(Box::new(SlateDBError::ManifestMissing));
+    };
+    Ok(serde_json::to_string(&manifest.core.checkpoints)?)
```

Comment:
+1 I think we should return a list of Checkpoints.

Reply:
Makes sense. We did it this way for the manifest because we didn't want to make all the internal metadata types public. I think it's reasonable for the `Checkpoint` struct to be public though since the user directly interacts with checkpoints (as opposed to say, the ssts).

Reply:
Wonder if we should create a `api.rs` or something eventually, to track all publicly visible structs/traits so we can do semver more easily...

Reply:
opened https://github.com/slatedb/slatedb/issues/335